lib/segment/src/vector_storage/simple_sparse_vector_storage.rs

```diff
@@ -151,7 +153,11 @@ impl VectorStorage for SimpleSparseVectorStorage {
     fn insert_vector(&mut self, key: PointOffsetType, vector: VectorRef) -> OperationResult<()> {
         let vector: &SparseVector = vector.try_into()?;
         debug_assert!(vector.is_sorted());
-        self.vectors.insert(key as usize, vector.clone());
+        self.vectors.resize_with(
```

Comment:
in normal times, I would ask for a unit test here, but as the memory representation it will go away I will spare you the work :)

Reply:
I just want to make https://github.com/qdrant/qdrant/pull/2802 simpler without internal stuff